packages/governance/src/contractGovernance/governApi.js

```diff
@@ -0,0 +1,132 @@
+
+import {
+  ChoiceMethod,
+  QuorumRule,
+  ElectionType,
+  coerceQuestionSpec,
+} from '../question.js';
+
+const { details: X } = assert;
+
+/**
+ * Make a pair of positions for a question about whether to invoke an API. If
+ * the vote passes, the method will be called on the governedApis facet with the
+ * parameters that were provided.
+ *
+ * @param {string} apiMethodName
+ * @param {[unknown]} methodParams
+ */
+const makeApiInvocationPositions = (apiMethodName, methodParams) => {
+  const positive = harden({ apiMethodName, methodParams });
+  const negative = harden({ dontInvoke: apiMethodName });
+  return { positive, negative };
+};
+
+/**
+ * manage contracts that allow governance to invoke functions.
+ *
+ * @param {ERef<ZoeService>} zoe
+ * @param {Instance} governedInstance
+ * @param {any} governedApis
+ * @param {string[]} governedNames
+ * @param {ERef<TimerService>} timer
+ * @param {() => Promise<PoserFacet>} getUpdatedPoserFacet
+ * @returns {Promise<ApiGovernor>}
+ */
+const setupApiGovernance = async (
+  zoe,
+  governedInstance,
+  governedApis,
+  governedNames,
+  timer,
+  getUpdatedPoserFacet,
+) => {
+  /** @type {WeakSet<Instance>} */
+  const voteCounters = new WeakSet();
+
+  /** @type {VoteOnApiInvocation} */
+  const voteOnApiInvocation = async (
+    apiMethodName,
+    methodParams,
+    voteCounterInstallation,
+    deadline,
+  ) => {
+    const outcomeOfUpdateP = makePromiseKit();
+
+    assert(
+      governedNames.includes(apiMethodName),
+      X`${apiMethodName} is not a governed API.`,
+    );
+
+    const { positive, negative } = makeApiInvocationPositions(
+      apiMethodName,
+      methodParams,
+    );
+
+    /** @type {ApiInvocationIssue} */
+    const issue = harden({ apiMethodName, methodParams });
+    const questionSpec = coerceQuestionSpec({
+      method: ChoiceMethod.UNRANKED,
+      issue,
+      positions: [positive, negative],
+      electionType: ElectionType.API_INVOCATION,
+      maxChoices: 1,
+      closingRule: { timer, deadline },
+      quorumRule: QuorumRule.MAJORITY,
+      tieOutcome: negative,
+    });
+
+    const { publicFacet: counterPublicFacet, instance: voteCounter } = await E(
+      getUpdatedPoserFacet(),
+    ).addQuestion(voteCounterInstallation, questionSpec);
+
+    voteCounters.add(voteCounter);
+
+    // CRUCIAL: Here we wait for the voteCounter to declare an outcome, and then
+    // attempt to invoke the API if that's what the vote called for. We need to
+    // make sure that outcomeOfUpdateP is updated whatever happens.
+    //
+    // * If the vote was negative, resolve to the outcome
+    // * If the API invocation succeeds, say so
+    // * If the update fails, reject the promise
+    // * if the vote outcome failed, reject the promise.
+    E(counterPublicFacet)
+      .getOutcome()
+      .then(outcome => {
+        if (keyEQ(positive, outcome)) {
+          E(governedApis)
+            [apiMethodName](...methodParams)
```

Comment:
What is to be invoked is defined in multiple places:
- `positive`
- `outcome`
- `issue`
- local scope from parameters of `voteOnApiInvocation`

I see a check that `outcome` and `positive` match, because that's a signal as to what the outcome was.

As an extra security check, and to make clear to the reader, please check here that the outcome matches the parameters. E.g.,
```suggestion
        if (keyEQ(positive, outcome)) {
          keyEQ(outcome, { apiMethodName, methodParams });
          E(governedApis)
            [apiMethodName](...methodParams)
```

Though as noted above, I think this must be `methodArgs`.

Reply:
Ahh!   I didn't catch that the comment on documentation also applied to the variable name in the code. I've changed the name and added the assertion.